### ISSUE TITLE
Icons in buttons need accessible names

### ISSUE THREAD

**Issue by gnomeontherun** [vmware/clarity#3530]
## Describe the bug

The documentation and demos do not properly setup buttons with icons so a screen reader can announce the button.

## How to reproduce

https://clarity.design/documentation/button-group

- Using a screen reader find the icons examples
- Read each icon button and observe that you do not hear any names for these controls

## Expected behavior

Buttons should have accessible names. Also documentation should tell people they cannot just put in icons without accessible names.

## Versions

**App**

* Angular: [e.g. 6]
* Node: [e.g. 8.10.0]
* Clarity: [e.g. 0.12.5]

**Device:**

* Type: [e.g. MacBook]
* OS: [e.g. iOS]
* Browser [e.g. chrome, safari]
* Version [e.g. 22]

## Additional notes

This applies to documentation for both buttons and button groups, worth searching all of the docs for any other icon buttons missing these details.
